### PR TITLE
Feature Importance to output

### PR DESCRIPTION
**Pull request by vaibhj**
E2E local test

https://ml.azure.com/experiments/id/4b4e34ef-331f-4b1a-b1b0-88615d6c45ad/runs/helpful_pear_svgb519mdm?wsid=/subscriptions/1aefdc5e-3a7c-4d71-a9f9-f5d3b03be19a/resourcegroups/TestingTeamProjectRG/providers/Microsoft.MachineLearningServices/workspaces/shrsha-eastus-test&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### CODE REVIEW

**Review by alanpo1 — COMMENTED** [review]
Can you link one of your e2e local tests to see

Added in description - VJ

**Comment by alanpo1** [comment]
Also there seems to be syntax errors

**Comment by ycheng35xo** [comment]
There are other signals using model_monitor_output_metrics components (feature_attribution_drift, prediction_drift), you need to change the yaml file for them as well.

File: assets/model_monitoring/components/model_monitor/model_monitor_output_metrics/spec.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ inputs:
     type: mltable
     mode: direct
     optional: true
+  feature_importance_all:
```
Comment (ycheng35xo) [inline assets/model_monitoring/components/model_monitor/model_monitor_output_metrics/spec.yaml:34]: Can you upgrade the component version? Also, you need to upgrade the signal version for all the signals using this component. You can refer to this PR https://github.com/Azure/azureml-assets/pull/2136/files#diff-9845970428817310efa54b26e7c1456a1d011f8e28327bfc6455120f6167e0d0

File: assets/model_monitoring/components/data_quality/data_quality_compute_metrics/spec.yaml [hunk 2]
```diff
@@ -60,3 +60,4 @@ args: >-
   --categorical_metrics ${{inputs.categorical_metrics}}
   --numerical_metrics ${{inputs.numerical_metrics}}
   --signal_metrics ${{outputs.signal_metrics}}
+  --feature_names ${{inputs.feature_names}}
```
[the remark this answers is not in the document]
Comment (vaibhj) [reply assets/model_monitoring/components/data_quality/data_quality_compute_metrics/spec.yaml:63]: This was bug.we need this input for right feature set.

File: assets/model_monitoring/components/src/model_monitor_feature_selector/selectors/feature_selector_top_by_attribution.py [hunk 3]
```diff
@@ -23,12 +23,21 @@ def select(
...
+        # Collect All feature and importance value from input
+        feature_importance_names_importance = (
+            self.feature_importance.select(self.feature_importance.feature, self.feature_importance.metric_value)
+            .collect()
         )
-        top_N_feature_importance_names = feature_importance_names[: self.N_value]
+        # Sort and find Top N features with higher importance
+        top_N_feature_importance_names_importance = sorted(
+            feature_importance_names_importance, key=lambda x: x[1], reverse=True)[: self.N_value]
+        print(top_N_feature_importance_names_importance)
```
Comment (ycheng35xo) [inline assets/model_monitoring/components/src/model_monitor_feature_selector/selectors/feature_selector_top_by_attribution.py:34]: Is this used for debugging? remove?

**Comment by ycheng35xo** [comment]
In the e2e test you shared, the new code you added was not tested. Please check the driver log and the output of the feature_importance component. I saw this below. There is no feature importance output for this job

[2024-01-23 00:02:55.779343] No target column given, creating an empty dataframe.

File: assets/model_monitoring/components/src/model_monitor_feature_selector/selectors/feature_selector_top_by_attribution.py (hunk 3 above)
Comment (vaibhj) [reply assets/model_monitoring/components/src/model_monitor_feature_selector/selectors/feature_selector_top_by_attribution.py:34]: yes it was for debugging .. I will remove

**Review by ycheng35xo — APPROVED** [review]

**Pull request merged**